### PR TITLE
calculate inflow and outflow energy and store them in energyTotals co…

### PR DESCRIPTION
**Pull request by adamstallard**
…llection

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.formatting.provider": "black"
```
Comment (adamstallard) [inline .vscode/settings.json:2]: Do we need this file in the repo?

Comment (0xoc) [reply .vscode/settings.json:2]: No. It should have been ignored.

File: scorer/verifications/aura.py [hunk 2]
```diff
@@ -4,18 +4,18 @@
 import time
 
 client = ArangoClient(hosts=config.ARANGO_SERVER)
-system = client.db('_system')
-snapshot = client.db('snapshot')
+system = client.db("_system")
+snapshot = client.db("snapshot")
```
Comment (adamstallard) [inline scorer/verifications/aura.py:8]: This python code base mostly uses single-quotes.  Maybe we need to put a `.editorconfig` file [like this one I created for Aura Backend](https://github.com/Meta-Node/aura-backend/blob/master/.editorconfig).  I'm pretty sure vscode will read `.editorconfig`

File: scorer/verifications/aura.py [hunk 3]
```diff
@@ -31,45 +31,50 @@
...
-    energy = snapshot.collection('energy')
-    energy_next = snapshot.collection('energyNext')
-    aura = snapshot.collection('aura')
+    if not snapshot.has_collection("energyTotals"):
+        snapshot.create_collection("energyTotals")
+
+    energy = snapshot.collection("energy")
+    energy_next = snapshot.collection("energyNext")
+    aura = snapshot.collection("aura")
+    enegry_totals = snapshot.collection("energyTotals")
```
Comment (adamstallard) [inline scorer/verifications/aura.py:57]:
```suggestion
    energy_totals = snapshot.collection("energyTotals")
```

File: scorer/verifications/aura.py [hunk 4]
```diff
@@ -31,45 +31,50 @@
...
 
     # Clear collections
 
     energy.truncate()
     energy_next.truncate()
     aura.truncate()
+    enegry_totals.truncate()
```
Comment (adamstallard) [inline scorer/verifications/aura.py:64]:
```suggestion
    energy_totals.truncate()
```

File: scorer/verifications/aura.py [hunk 5]
```diff
@@ -111,16 +119,17 @@ def verify(block):
                         in energyNext
                         insert { _from: ea._from, _to: ea._to, energy: energySent, timestamp: @timestamp }
                         in energyFlow
-            ''', bind_vars={
-                "timestamp": timestamp
-            })
+            """,
+                bind_vars={"timestamp": timestamp},
```
Comment (adamstallard) [inline scorer/verifications/aura.py:123]: I prefer the other formatting to make clear what's in the object.

File: scorer/verifications/aura.py (hunk 2 above)
Comment (0xoc) [reply scorer/verifications/aura.py:8]: I like using a standard formatter (in this case ```black```) to format the code automatically whenever I save the file. By default ```black``` replaces all double quotes with single quotes. And I guess we can change that in ```.editorconfig```. or remove ```black``` altogether if you don't like it.

File: scorer/verifications/aura.py [hunk 6]
```diff
@@ -132,20 +141,44 @@ def verify(block):
...
-        energy = snapshot.collection('energy')
-        energy_next = snapshot.collection('energyNext')
+        energy.rename("energyTemp")
+        energy_next.rename("energy")
+        energy.rename("energyNext")
+        energy = snapshot.collection("energy")
+        energy_next = snapshot.collection("energyNext")
+
+    # Energy flow table is clean now
+    # use this data to comput total energy flow to each aura player
```
Comment (adamstallard) [inline scorer/verifications/aura.py]:
```suggestion
    # Use this data to compute the total energy flow to each Aura player.
```

File: scorer/verifications/aura.py [hunk 7]
```diff
@@ -132,20 +141,44 @@ def verify(block):
...
+    # we need to sum all row with the same _to and _timestamp to get total inflow energy
```
Comment (adamstallard) [inline scorer/verifications/aura.py:157]:
```suggestion
    # We need to sum all the rows with the same _to and _timestamp to get the total energy inflow.
```

**Review by adamstallard — COMMENTED** [review]
Looks good @spsina .  (I suggested changes.)  I haven't run the aql query, but it looks right to me.

File: scorer/verifications/aura.py (hunk 2 above)
Comment (adamstallard) [reply scorer/verifications/aura.py:8]: Not everyone uses vscode.  I'm not the only one who has contributed to this repo over the years.  I think we should keep with the formatting we've been using for years unless there's something wrong with it.  (There isn't.  It's all compatible with PEP-8.)

Comment (0xoc) [reply scorer/verifications/aura.py:8]: Ok. I will remove it.

Comment (adamstallard) [reply scorer/verifications/aura.py:8]: > By default `black` replaces all double quotes with single quotes.

I think you mean the opposite, right? It prefers double quotes over single quotes?

If we could create an `.editorconfig` (or something similar--like "black" [though I've never used it]) that can match the styles we're already using, that would be perfect.

Actually, that would be good to merge back into the main BrightID Node repo.  Whoever started on the python code originally never created a style config file for editors.  It's nice to have.

I'm not familiar with "black" so I don't know if most IDEs or editors can read it.
If so, then great, we can use it.

But this is a tool that BrightID-node was missing from the beginning, so if we can find the config that matches their style, that would be cool, but might be too much work to be worth it.  I don't know.

**Pull request closed without merging**